### PR TITLE
Feature/get accounts

### PR DESCRIPTION
**Pull request by TinyMurky**
### DEVELOP
- add get account
- add get accounts

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 
- new Class / Component: 0
- new loop: 1
    - format accounts to IAccount[]
- new recursive: 0
- high risk: 0
- new sql: 2
    - read account
    - read get accounts

### UML

- none

### CODE REVIEW

File: src/lib/utils/type_guard/account.ts [hunk 1]
```diff
@@ -13,6 +14,11 @@ export function isEventType(data: any): data is EventType {
   return Object.values(EventType).includes(data as EventType);
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isAccountType(data: any): data is AccountType {
+  return Object.values(AccountType).includes(data as AccountType);
```
Comment (Luphia) [inline src/lib/utils/type_guard/account.ts:19]: - do not calculate in return

File: src/pages/api/v1/company/[companyId]/account/[accountId]/index.test.ts [hunk 2]
```diff
@@ -49,61 +71,39 @@ describe('updateOwnAccountInfoById API Handler Tests', () => {
           account: expect.any(String),
           code: expect.any(String),
           name: expect.any(String),
+          createdAt: expect.any(Number),
+          updatedAt: expect.any(Number),
```
Comment (Luphia) [inline src/pages/api/v1/company/[companyId]/account/[accountId]/index.test.ts:75]: - do not calculate in function input

File: src/pages/api/v1/company/[companyId]/account/[accountId]/index.test.ts [hunk 3]
```diff
@@ -49,61 +71,39 @@ describe('updateOwnAccountInfoById API Handler Tests', () => {
...
+    req.query = { companyId: '1', accountId: 'a' };
     await handler(req, res);
     expect(res.status).toHaveBeenCalledWith(422);
     expect(res.json).toHaveBeenCalledWith(
       expect.objectContaining({
         powerby: expect.any(String),
-        success: expect.any(Boolean),
+        success: false,
         code: expect.stringContaining('422'),
         message: expect.any(String),
```
Comment (Luphia) [inline src/pages/api/v1/company/[companyId]/account/[accountId]/index.test.ts:106]: - do not calculate in function input

File: src/pages/api/v1/company/[companyId]/account/index.test.ts [hunk 4]
```diff
@@ -75,44 +110,58 @@ describe('getOwnAccounts API Handler Tests', () => {
...
         message: expect.any(String),
-        payload: expect.objectContaining({
-          id: expect.any(Number),
-          type: expect.any(String),
-          liquidity: expect.any(Boolean),
-          account: expect.any(String),
-          code: expect.any(String),
-          name: expect.any(String),
-        }),
+        payload: expect.any(Object),
```
Comment (Luphia) [inline src/pages/api/v1/company/[companyId]/account/index.test.ts:125]: - do not calculate in function input

File: src/pages/api/v1/company/[companyId]/account/index.test.ts [hunk 5]
```diff
@@ -75,44 +110,58 @@ describe('getOwnAccounts API Handler Tests', () => {
...
+    req.query = { companyId: '1', type: 'asset', liquidity: 'invalid', page: '1', limit: '10' };
+    await handler(req, res);
+    expect(res.status).toHaveBeenCalledWith(422);
+    expect(res.json).toHaveBeenCalledWith(
+      expect.objectContaining({
+        powerby: expect.any(String),
+        success: expect.any(Boolean),
+        code: expect.stringContaining('422'),
+        message: expect.any(String),
+        payload: expect.any(Object),
```
Comment (Luphia) [inline src/pages/api/v1/company/[companyId]/account/index.test.ts:141]: - do not calculate in function input

File: src/pages/api/v1/company/[companyId]/account/index.test.ts [hunk 6]
```diff
@@ -75,44 +110,58 @@ describe('getOwnAccounts API Handler Tests', () => {
...
+    req.query = { companyId: '1', type: 'asset', liquidity: 'true', page: 'abc', limit: '10' };
+    await handler(req, res);
+    expect(res.status).toHaveBeenCalledWith(422);
+    expect(res.json).toHaveBeenCalledWith(
+      expect.objectContaining({
+        powerby: expect.any(String),
+        success: expect.any(Boolean),
+        code: expect.stringContaining('422'),
+        message: expect.any(String),
+        payload: expect.any(Object),
```
Comment (Luphia) [inline src/pages/api/v1/company/[companyId]/account/index.test.ts:157]: - do not calculate in function input

**Review by Luphia — CHANGES_REQUESTED** [review]

File: src/pages/api/v1/company/[companyId]/account/index.test.ts (hunk 6 above)
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/account/index.test.ts:157]: done

File: src/pages/api/v1/company/[companyId]/account/index.test.ts (hunk 4 above)
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/account/index.test.ts:125]: done

File: src/pages/api/v1/company/[companyId]/account/[accountId]/index.test.ts (hunk 2 above)
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/account/[accountId]/index.test.ts:75]: done

File: src/lib/utils/type_guard/account.ts (hunk 1 above)
Comment (TinyMurky) [reply src/lib/utils/type_guard/account.ts:19]: done

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**